WeatherMonitoringSystem/Domain/Utilities/UserInteraction.cs

```diff
@@ -0,0 +1,140 @@
+﻿
+using System.Xml;
+using WeatherMonitoringSystem.Bots;
+using WeatherMonitoringSystem.Core;
+using WeatherMonitoringSystem.DataParsers;
+using Newtonsoft.Json;
+
+namespace WeatherMonitoringSystem.UI;
+
+public static class UserInteraction
+{
+
+    public static void Run()
+    {
+        PrintMessage("Welcome to the Real-Time Weather Monitoring System!");
+
+        WeatherParserStrategy parserStrategy = new WeatherParserStrategy();
+        IWeatherDataParser parser = null;
+
+        while (true)
+        {
+            string format = GetDataFormat();
+
+            try
+            {
```

Comment:
you can just do one big try with all code in while loop, and multiple catch blocks